src/main/java/com/exadel/discount/controller/AuthenticationController.java

```diff
@@ -0,0 +1,59 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.dto.authentication.RefreshResponse;
+import com.exadel.discount.security.annotation.RefreshAccess;
+import com.exadel.discount.dto.authentication.AuthenticationRequest;
+import com.exadel.discount.dto.authentication.AuthenticationResponse;
+import com.exadel.discount.service.JwtGenerationService;
+import com.exadel.discount.service.impl.UserDetailsServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/authenticate")
+@RequiredArgsConstructor
+public class AuthenticationController {
+    private final AuthenticationManager authenticationManager;
+    private final UserDetailsServiceImpl userDetailsService;
+    private final JwtGenerationService jwtGenerationService;
+
+
+    @PostMapping("/login")
+    public ResponseEntity<AuthenticationResponse> createAuthenticationToken(@RequestBody AuthenticationRequest request) {
+        authenticationManager.authenticate(
```

Comment:
Do you relly need ResponseEntity here?

Reply:
You don't need ResponseEntity! And I have already mentioned about it: https://github.com/SergeyKoval/2021_june_team4_be/pull/28#discussion_r661239752